### PR TITLE
Backports 0.15 pr21

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: doc/REST-interface.md [hunk 1]
```diff
@@ -57,25 +59,25 @@ Example:
...
             "hex" : "76a9141c7cebb529b86a04c683dfa87be49de35bcf589e88ac",
             "reqSigs" : 1,
-            "asm" : "OP_DUP OP_HASH160 1c7cebb529b86a04c683dfa87be49de35bcf589e OP_EQUALVERIFY OP_CHECKSIG"
-         },
-         "value" : 8.8687,
-         "height" : 2147483647,
-         "txvers" : 1
+            "type" : "pubkeyhash",
+            "addresses" : [
+               "mi7as51dvLJsizWnTMurtRmrP8hG2m1XvD"
```
Comment (UdjinM6) [inline doc/REST-interface.md]: should probably dashify this

File: doc/REST-interface.md [hunk 2]
```diff
@@ -87,6 +89,8 @@ Only supports JSON as output format.
 * size : (numeric) the number of transactions in the TX mempool
 * bytes : (numeric) size of the TX mempool in bytes
 * usage : (numeric) total TX mempool memory usage
+* maxmempool : (numeric) maximum memory usage for the mempool in bytes
+* mempoolminfee : (numeric) minimum feerate (BTC per KB) for tx to be accepted
```
Comment (UdjinM6) [inline doc/REST-interface.md]: s/BTC/DASH/

File: src/wallet/wallet.cpp [hunk 3]
```diff
@@ -2083,7 +2083,7 @@ CAmount CWalletTx::GetAvailableCredit(bool fUseCache) const
             const CTxOut &txout = tx->vout[i];
             nCredit += pwallet->GetCredit(txout, ISMINE_SPENDABLE);
             if (!MoneyRange(nCredit))
-                throw std::runtime_error(std::string(__func__) + ": value out of range");
+                throw std::runtime_error(std::string(__func__) + " : value out of range");
```
Comment (UdjinM6) [inline src/wallet/wallet.cpp]: this looks like a mistake, I would revert it

File: src/rpc/misc.cpp [hunk 4]
```diff
@@ -335,6 +335,14 @@ UniValue validateaddress(const JSONRPCRequest& request)
             "  \"ismine\" : true|false,        (boolean) If the address is yours or not\n"
             "  \"iswatchonly\" : true|false,   (boolean) If the address is watchonly\n"
             "  \"isscript\" : true|false,      (boolean) If the key is a script\n"
+            "  \"script\" : \"type\"             (string, optional) The output script type. Possible types: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata, witness_v0_keyhash, witness_v0_scripthash\n"
```
Comment (UdjinM6) [inline src/rpc/misc.cpp]:
```suggestion
            "  \"script\" : \"type\"             (string, optional) The output script type. Possible types: nonstandard, pubkey, pubkeyhash, scripthash, multisig, nulldata\n"
```

File: contrib/verifybinaries/verify.sh [hunk 5]
```diff
@@ -3,7 +3,8 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-###   This script attempts to download the signature file SHA256SUMS.asc from bitcoin.org
+###   This script attempts to download the signature file SHA256SUMS.asc from
+###   bitcoincore.org and bitcoin.org and compares them.
```
Comment (UdjinM6) [inline contrib/verifybinaries/verify.sh]: .. which makes no sense to do in our case but it's probably ok :)

File: configure.ac [hunk 6]
```diff
@@ -1344,4 +1350,5 @@ echo "  CPPFLAGS            = $CPPFLAGS"
 echo "  CXX                 = $CXX"
 echo "  CXXFLAGS            = $CXXFLAGS"
 echo "  LDFLAGS             = $LDFLAGS"
+echo "  ARFLAGS       = $ARFLAGS"
```
Comment (UdjinM6) [inline configure.ac]:
```suggestion
echo "  ARFLAGS             = $ARFLAGS"
```

File: src/bench/base58.cpp [hunk 7]
```diff
@@ -7,34 +7,37 @@
 #include "validation.h"
 #include "base58.h"
 
+#include <array>
```
Comment (UdjinM6) [inline src/bench/base58.cpp]: `verify_script.cpp` is not in our repo. Is yet another earlier PR missing?

File: test/functional/test_framework/util.py [hunk 8]
```diff
@@ -601,7 +601,10 @@ def satoshi_round(amount):
 # Helper to create at least "count" utxos
 # Pass in a fee that is sufficient for relay and mining new transactions.
 def create_confirmed_utxos(fee, node, count):
-    node.generate(int(0.5*count)+101)
+    to_generate = int(0.5*count)+101
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply test/functional/test_framework/util.py]: yup, depends on #10148 which I was having some problems getting to work and decided to go back to. Dropped #10704, to be merged after #10148

File: doc/REST-interface.md (hunk 1 above)
Comment (PastaPastaPasta) [reply doc/REST-interface.md]: what type of address should it be? :see_no_evil: I'm not too familiar with bitcoin's address types

File: doc/REST-interface.md (hunk 2 above)
Comment (PastaPastaPasta) [reply doc/REST-interface.md]: resolved

File: src/wallet/wallet.cpp (hunk 3 above)
Comment (PastaPastaPasta) [reply src/wallet/wallet.cpp]: dropped

File: src/rpc/misc.cpp (hunk 4 above)
Comment (PastaPastaPasta) [reply src/rpc/misc.cpp]: resolved

File: doc/REST-interface.md (hunk 1 above)
Comment (UdjinM6) [reply doc/REST-interface.md]: that's a testnet/regtest p2pkh

File: src/bench/base58.cpp (hunk 7 above)
Comment (PastaPastaPasta) [reply src/bench/base58.cpp]: added in https://github.com/bitcoin/bitcoin/pull/8873 @codablock backported most of this pr so I assume there was a reason he dropped this file. I don't see any mention of such in https://github.com/dashpay/dash/pull/1849 however. File seems (maybe) witness/segwit dependant

File: doc/REST-interface.md (hunk 1 above)
Comment (UdjinM6) [reply doc/REST-interface.md]: Actually, this file need to be dashified as a whole. Let's ignore it for now then.

File: doc/REST-interface.md [hunk 9]
```diff
@@ -73,7 +73,7 @@ $ curl localhost:18332/rest/getutxos/checkmempool/b2cdfd7b89def827ff8af7cd9bff76
             "reqSigs" : 1,
             "type" : "pubkeyhash",
             "addresses" : [
-               "mi7as51dvLJsizWnTMurtRmrP8hG2m1XvD"
+               "8hVtrkizS3dQbTnPiCVdV7nFuvoce1gJsj"
```
Comment (UdjinM6) [inline doc/REST-interface.md]: that's a p2sh address, not a p2pkh one ;) also see, https://github.com/dashpay/dash/pull/3029#discussion_r304692685 - let's postpone dashification of this file

Comment (PastaPastaPasta) [reply doc/REST-interface.md]: What are you talking about? I see no such commit adding that weird foreign address!

**Pull request merged**